packetio/buffer.go

```diff
@@ -190,13 +190,19 @@ func (b *Buffer) WriteContext(ctx context.Context, packet []byte) (int, error) {
 // Returns io.EOF if the buffer is closed.
 func (b *Buffer) Read(packet []byte) (n int, err error) {
 	// Return immediately if the deadline is already exceeded.
-	select {
-	case <-b.readDeadline.Done():
```

Comment:
Read without deadline check at the beginning lacks compatibility with `net.Conn` as it may return data after deadline exceed.

Reply:
To make the comparison fair, check of `<-ctx.Done()` should also be omitted.